fex2/preconfigured/experiments/common.sh

```diff
@@ -53,54 +78,44 @@ BUILD_DIR=${BUILD_DIR:-${BUILD_ROOT}/${NAME}}
 IFS=' ' read -r -a types <<<"$BUILD_TYPES"
 IFS=' ' read -r -a thread_counts <<<"$NUM_THREADS"
 
-is_benchmark_suite=0
-command=""
-execution_header=()
-input=""
-bin=""
-
-declare -a execution_header
+# These values are set in the run script of each experiment
+fex2_argument_is_benchmark_suite=false
+fex2_argument_benchmarks=""
+fex2_argument_command=""
 
 # shell output colors
-if [ $COLORED_LOGS -eq 1 ]; then
+if [ $COLORED_LOGS == true ]; then
     DEBUG='\033[94m\033[1m'
     WARNING='\033[93m\033[1m'
     ERROR='\033[91m\033[1m'
     ENDC='\033[0m'
-else
-    DEBUG=''
-    WARNING=''
-    ERROR=''
-    ENDC=''
 fi
 
 ##############################
 # Shortcuts and helpers
 ##############################
-function header_push() {
-    execution_header+=("$1")
-}
-
-function header_pop() {
-    local top=("${!execution_header[@]}")
-    local top_index=${#execution_header[@]}
-    if [ "$top_index" -le 0 ]; then
-        echo "Current Header: " "${execution_header[@]}"
-        error_exit "header_pop: unbalanced pop from the header" 1
-    fi
-    unset 'execution_header[${top[@]: -1}]'
+function is() {
+    local value=$1
+    if [[ "$value" == true ]]; then
+        return 0;
+    elif [[ "$value" == false ]]; then
+        return 1;
+    else
+        unexpected_error_exit "Invalid boolean value $value" 1
+    fi;
```

Comment:
Actually, I find this function quite handy and defensive against unwanted values in your bool variables. Why not extend the vague bash with some solid, defensive functions? :smile:

Reply:
Because it's not standard bash and I want to keep the learning curve of a new Fex user as flat as possible.

Let's just agree on the following: If you want to use some non-vanila-bash feature, check the code base of Linux (https://github.com/torvalds/linux). If you can find similar patterns in it, you are free to use it. Otherwise, better not.